DO NOT MERGE (#4724)
remix-run/remix: packages/remix-server-runtime/CHANGELOG.md

# `@remix-run/server-runtime`

## 1.8.0

### Minor Changes

> - We have been busy at work [Layering Remix on top of React Router 6.4](https://github.com/remix-run/remix/blob/main/decisions/0007-remix-on-react-router-6-4-0.md) and are excited to be releasing the step 1 in this process that consists of running the server flow through a local copy the new framework agnostic `@remix-run/router`. ([#4612](https://github.com/remix-run/remix/pull/4612))

Reviewer: ```suggestion
- We have been busy at work [Layering Remix on top of React Router 6.4](https://github.com/remix-run/remix/blob/main/decisions/0007-remix-on-react-router-6-4-0.md) and are excited to be releasing step 1 in this process that consists of running the server flow through a local copy the new framework agnostic `@remix-run/router`. ([#4612](https://github.com/remix-run/remix/pull/4612))
```

- Importing functions and types from the `remix` package is deprecated, and all ([#3284](https://github.com/remix-run/remix/pull/3284))
  exported modules will be removed in the next major release. For more details,
  [see the release notes for 1.4.0](https://github.com/remix-run/remix/releases/tag/v1.4.0)
  where these changes were first announced.
- Added support for a new route `meta` API to handle arrays of tags instead of an object. For details, check out the [RFC](https://github.com/remix-run/remix/discussions/4462). ([#4610](https://github.com/remix-run/remix/pull/4610))

### Patch Changes

> - Update to use stable `@remix-run/router@1.0.4` release ([#4720](https://github.com/remix-run/remix/pull/4720))

Reviewer: I'd remove this bullet since it's addressed in the minor bullet above, the changeset was just needed to trigger a new prerelease

> - Remove duplicated router code in favor of `@remix-run/router` dependency ([#4699](https://github.com/remix-run/remix/pull/4699))

Reviewer: Also can get rid of this one.  If we care to keep the PR references we can add them to the minor RRR bullet above

- Properly categorize internal framework-thrown error Responses as error boundary errors ([#4385](https://github.com/remix-run/remix/pull/4385))

  Previously there was some ambiguity around _"thrown Responses go to the `CatchBoundary`"_.
  The `CatchBoundary` exists to give the _user_ a place to handle non-happy path code flows

>   such that they can throw Response instances from _their own code_ and handle them in a

Reviewer: ```suggestion
  such that they can throw `Response` instances from _their own code_ and handle them in a
```

  `CatchBoundary`. However, there are a handful of framework-internal errors that make

>   sense to have a non-500 status code, and the fact that these were being thrown as Responses

Reviewer: ```suggestion
  sense to have a non-500 status code, and the fact that these were being thrown as `Response` instances
```

>   was causing them to go into the CatchBoundary, even though they were not user-thrown.

Reviewer: ```suggestion
  was causing them to go into the `CatchBoundary`, even though they were not user-thrown.
```

  With this change, anything thrown by the framework itself (`Error` or `Response`) will
  go to the `ErrorBoundary`, and any user-thrown `Response` instances will go to the

>   `CatchBoundary`. Thereis one exception to this rule, which is that framework-detected

Reviewer: ```suggestion
  `CatchBoundary`. There is one exception to this rule, which is that framework-detected
```
